### PR TITLE
Make users-permissions auth strategy use the content API permissions engine

### PR DESCRIPTION
**Pull request by Convly**
### What does it do?

- **Authentication:** users-permissions auth strategy now generate an ability using the content API permission engine instance and pass it down to the auth context for every authenticated request
- **Authorization:** users-permissions auth strategy now use the previously generated ability to check if the request can be allowed or not 

### Why is it needed?

It's the first step in harmonizing every content API auth layer by using the shared content API permission engine to make the checks. 

### How to test it?

Try to make authenticated (and non-authenticated) request to the content API as usual; nothing should've changed. This PR should only change the implementation, not the behavior.

### CODE REVIEW

File: packages/plugins/users-permissions/server/services/permission.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+   *
+   * const permission = toContentAPIPermission(upPermission);
+   * // ^? { action: 'find', subject: 'api::foo.foo' }
+   *
+   * @param {object} permission
+   * @param {string} permission.action
+   *
+   * @return {{ action: string, subject: string }}
+   */
+  toContentAPIPermission(permission) {
```
Comment (Convly) [inline packages/plugins/users-permissions/server/services/permission.js:46]: Still not sure 100% how we should treat users-permissions actions.

They're built using the controller's method name as action and the` source::api.controller` as subject (everything is then concatenated into a unique action name). 

For now I've let them as they're everywhere except when registering them into the content API permissions engine + when generating & testing abilities (in the auth strategy)

**Comment by derrickmehaffy** [comment]
Yooooo this is a great first step! So excited to see where this goes  😎

**Review by innerdvations — APPROVED** [review]

File: packages/plugins/users-permissions/server/services/permission.js (hunk 1 above)
Comment (Convly) [reply packages/plugins/users-permissions/server/services/permission.js:40]: Decided to stay consistent with how they were generated previously, but we'll need to think about it when we'll create the API token bridge with the content API engine.

**tl;dr: for now we're keeping action only**
